src/invariant/ArchivedStateConsistency.cpp

```diff
@@ -30,7 +30,8 @@ ArchivedStateConsistency::ArchivedStateConsistency() : Invariant(true)
 std::string
 ArchivedStateConsistency::checkSnapshot(
     CompleteConstLedgerStatePtr ledgerState,
-    InMemorySorobanState const& inMemorySnapshot)
```

Comment:
This is also a long running test, so we should check `isStopping` in the `scanForEntriesOfType` lambda and return `Loop::COMPLETE` early if we encounter it.